### PR TITLE
Use port names and numbers for filter match

### PR DESCRIPTION
**Pull request by rshriram**
Signed-off-by: Shriram Rajagopalan <shriramr@vmware.com>

### CODE REVIEW

**Comment by rshriram** [comment]
@spikecurtis

**Comment by rshriram** [comment]
@ostromart

File: networking/v1alpha3/envoy_filter.proto [hunk 1]
```diff
@@ -85,12 +90,21 @@ message EnvoyFilters {
     // matches all listeners.
     ListenerType listener_type = 2;
 
+    enum ListenerProtocol {
+      // All protocols
+      ALL = 0;
+      // HTTP or HTTPS (with termination) / HTTP2/gRPC
+      HTTP = 1;
+      // Any non-HTTP listener
+      TCP = 2;
```
Comment (spikecurtis) [inline networking/v1alpha3/envoy_filter.proto]: did we drop Mongo?

Comment (rshriram) [reply networking/v1alpha3/envoy_filter.proto]: Yeah.. this is simply saying its HTTP/TCP.

**Pull request merged**